tests/Tests.AzureAppConfiguration/Integration/IntegrationTests.cs

```diff
@@ -0,0 +1,2041 @@
+                {
+                    // Check if the setting is older than the threshold
+                    if (setting.LastModified < cutoffTime)
+                    {
+                        configSettingsToCleanup.Add(setting);
+                        staleConfigCount++;
+                    }
+                }
+
+                foreach (ConfigurationSetting setting in configSettingsToCleanup)
+                {
+                    cleanupTasks.Add(_configClient.DeleteConfigurationSettingAsync(setting.Key, setting.Label));
+                }
+
+                int staleSnapshotCount = 0;
+                AsyncPageable<ConfigurationSnapshot> snapshots = _configClient.GetSnapshotsAsync(new SnapshotSelector());
+                await foreach (ConfigurationSnapshot snapshot in snapshots)
+                {
+                    if (snapshot.Name.StartsWith("snapshot-" + TestKeyPrefix) && snapshot.CreatedOn < cutoffTime)
+                    {
+                        cleanupTasks.Add(_configClient.ArchiveSnapshotAsync(snapshot.Name));
+                        staleSnapshotCount++;
+                    }
+                }
+
+                int staleSecretCount = 0;
+                if (_secretClient != null)
+                {
+                    AsyncPageable<SecretProperties> secrets = _secretClient.GetPropertiesOfSecretsAsync();
+                    await foreach (SecretProperties secretProperties in secrets)
+                    {
+                        if (secretProperties.Name.StartsWith(TestKeyPrefix) && secretProperties.CreatedOn.HasValue && secretProperties.CreatedOn.Value < cutoffTime)
+                        {
+                            cleanupTasks.Add(_secretClient.StartDeleteSecretAsync(secretProperties.Name));
+                            staleSecretCount++;
+                        }
+                    }
+                }
+
+                // Wait for all cleanup tasks to complete
+                await Task.WhenAll(cleanupTasks);
+
+                Console.WriteLine($"Cleaned up {staleConfigCount} stale configuration settings, {staleSnapshotCount} snapshots, and {staleSecretCount} secrets");
+            }
+            catch (RequestFailedException ex)
+            {
+                Console.WriteLine($"Error during stale resource cleanup: {ex.Message}");
+                // Continue execution even if cleanup fails
+            }
+        }
+
+        public async Task DisposeAsync()
+        {
+            await CleanupStaleResources();
+        }
+
+        [Flags]
+        private enum TestDataTypes
+        {
+            None = 0,
+            KeyValues = 1,
+            FeatureFlags = 2,
+            KeyVaultReferences = 4,
+            TaggedSettings = 8,
+            All = KeyValues | FeatureFlags | KeyVaultReferences
+        }
+
+        private async Task<TestContext> SetupTestKeys(string testName, TestDataTypes dataTypesToCreate = TestDataTypes.All)
```

Comment:
Creating a type adds a bit more complexity then we usually aim for. I'd suggest updating SetupTestKeys a bit.

Suggestion

```
TestContext context = await SetupTestKeys("footest", TestDataTypes.KeyValues | TestDataTypes.FeatureFlags)
```

becomes

```
TestContext context = await CreateTestContext("footest");

await SetupFeatureFlags(context);

await SetupTestKeys(context);
```

Reply:
Makes sense, removed the flag and created a method for each flag value that existed instead.